### PR TITLE
Shade re2j dependency (used by gRPC)

### PR DESCRIPTION
**Pull request by singhravidutt**
For debian11 re2j seems to me missing as it's a required jar for connector shading it.

```
 23/02/15 19:05:02 ERROR ManagedChannelImpl: [Channel<1>: (google-c2p-experimental:///storage.googleapis.com)] Uncaught exception in the SynchronizationContext. Panic!
 java.lang.NoClassDefFoundError: com/google/re2j/PatternSyntaxException
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.xds.ClientXdsClient.<init>(ClientXdsClient.java:102) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.xds.SharedXdsClientPoolProvider$RefCountedXdsClientObjectPool.getObject(SharedXdsClientPoolProvider.java:126) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.xds.SharedXdsClientPoolProvider$RefCountedXdsClientObjectPool.getObject(SharedXdsClientPoolProvider.java:103) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.xds.XdsNameResolver.start(XdsNameResolver.java:186) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.googleapis.GoogleCloudToProdNameResolver$1Resolve$1.run(GoogleCloudToProdNameResolver.java:162) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.SynchronizationContext.drain(SynchronizationContext.java:95) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.SynchronizationContext.execute(SynchronizationContext.java:127) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at com.google.cloud.hadoop.repackaged.gcs.io.grpc.googleapis.GoogleCloudToProdNameResolver$1Resolve.run(GoogleCloudToProdNameResolver.java:157) ~[gcs-connector-hadoop3-2.2.11.jar:?]
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
 	at java.lang.Thread.run(Thread.java:829) ~[?:?]
 Caused by: java.lang.ClassNotFoundException: com.google.re2j.PatternSyntaxException
 	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
 	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
 	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
 	... 11 more
```

### CODE REVIEW

File: gcs/pom.xml [hunk 1]
```diff
@@ -253,6 +254,7 @@
                     <include>com.google.storage.**</include>
                     <include>com.google.thirdparty.**</include>
                     <include>com.google.type.**</include>
+                    <include>com.google.re2j.**</include>
```
Comment (arunkumarchacko) [inline gcs/pom.xml:257]: keep this sorted here as well.

File: gcs/pom.xml [hunk 2]
```diff
@@ -220,6 +220,7 @@
                   <include>com.google.http-client</include>
                   <include>com.google.oauth-client</include>
                   <include>com.google.protobuf</include>
+                  <include>com.google.re2j</include>
```
Comment (arunkumarchacko) [inline gcs/pom.xml:223]: is this introduced by Veneer?

Comment (singhravidutt) [reply gcs/pom.xml:223]: I has nothing to do with java-storage. Updated Summary of PR.

File: gcs/pom.xml (hunk 1 above)
Comment (singhravidutt) [reply gcs/pom.xml]: Done.

File: gcs/pom.xml [hunk 3]
```diff
@@ -250,6 +251,7 @@
                     <include>com.google.longrunning.**</include>
                     <include>com.google.protobuf.**</include>
                     <include>com.google.rpc.**</include>
+                    <include>com.google.re2j.**</include>
```
Comment (medb) [inline gcs/pom.xml:254]: nit: sort alphabetically

File: gcs/pom.xml (hunk 2 above)
Comment (medb) [reply gcs/pom.xml:223]: So what GCS connector dependency uses `re2j`? Is there a stack trace for the failure that this PR fixes?

**Review by medb — CHANGES_REQUESTED** [review]

File: gcs/pom.xml (hunk 2 above)
Comment (singhravidutt) [reply gcs/pom.xml:223]: It's getting used in gRPC DirectPath. Updated PR description with stack trace.

**Pull request merged**